platform/mbed_assert.h

```diff
@@ -63,7 +64,7 @@ void mbed_assert_internal(const char *expr, const char *file, int line);
 #define MBED_ASSERT(expr)                                \
 do {                                                     \
     if (!(expr)) {                                       \
-        mbed_assert_internal(#expr, __FILE__, __LINE__); \
+        MBED_CRIT("ASRT", "mbed assertation failed: %s \n", #expr); \
```

Comment:
Would it be possible to do the same with `error` ?

Reply:
Error / Critical error should be distinguished. Not all errors are critical hence we have moved application termination from error to critical.

Or did you meant dumping Filename and Line number? Other teams have very strong views of not adding filename or line number.